File: src/algorithms/move-zeroes.js
/**
 * Given an integer array nums, moves all 0's to the end of it while maintaining
 * the relative order of the non-zero elements. Returns the modified array.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number[]} - Returns a modified array
 */
function moveZeroes (nums) {
  let i = 0;
  for (const num of nums) {
    if (num !== 0) {
      nums[i] = num;
      i++;
    }
  }
  while (i < nums.length) {
    nums[i] = 0;
    i++;
  }
  return nums;
}

File: src/algorithms/move-zeroes.py
def moveZeroes(nums):
    """Given an integer array nums, move all 0's to the end of it while 
    maintaining the relative order of the non-zero elements. Note that you must 
    do this in-place without making a copy of the array.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        list[int]: Returns a modified list of integers.
    """
    i = 0
    for j in range(len(nums)):
        if nums[j] != 0:
            nums[j], nums[i] = nums[i], nums[j]
            i += 1
    return nums
